src/main/java/com/conveyal/r5/transit/TransportNetworkCache.java

```diff
@@ -121,16 +121,29 @@ public synchronized TransportNetwork getNetworkForScenario (String networkId, Pr
             Scenario scenario;
             if (request.scenario == null && request.scenarioId != null) {
                 // resolve scenario
-                LOG.info("Retrieving scenario from S3");
-                // TODO offline mode
-                S3Object obj = s3.getObject(sourceBucket, String.format("%s_%s.json", networkId, scenarioId));
-                InputStream is = obj.getObjectContent();
+                LOG.info("Retrieving scenario stored separately");
```

Comment:
Not clear what "separately" means. If I understand correctly this is a multilevel cache where cache misses on the local filesystem fall through to S3.